### PR TITLE
Filesystem API: Add compressed file size check to PclZip.

### PR DESCRIPTION
**Pull request by yakimun**
These changes fix the fatal error when the compressed file has a size of 0.

Trac ticket: https://core.trac.wordpress.org/ticket/52018

### CODE REVIEW

File: src/wp-admin/includes/class-pclzip.php [hunk 1]
```diff
@@ -4209,12 +4209,20 @@ function privExtractFileAsString(&$p_entry, &$p_string, &$p_options)
         if ($p_entry['compression'] == 0) {
 
           // ----- Reading the file
-          $p_string = @fread($this->zip_fd, $p_entry['compressed_size']);
+          if ($p_entry['compressed_size'] > 0) {
+            $p_string = @fread($this->zip_fd, $p_entry['compressed_size']);
+          } else {
+            $p_string = false;
```
Comment (jrfnl) [inline src/wp-admin/includes/class-pclzip.php:4215]: This adds a different and unexpected new type for this variable (boolean, while string is expected), while this is a variable being changed by reference.

Suggestion:
```php
            $p_string = '';
```

File: src/wp-admin/includes/class-pclzip.php [hunk 2]
```diff
@@ -4209,12 +4209,20 @@ function privExtractFileAsString(&$p_entry, &$p_string, &$p_options)
...
+          }
         }
         else {
 
           // ----- Reading the file
-          $v_data = @fread($this->zip_fd, $p_entry['compressed_size']);
+          if ($p_entry['compressed_size'] > 0) {
+            $v_data = @fread($this->zip_fd, $p_entry['compressed_size']);
+          } else {
+            $v_data = false;
```
Comment (jrfnl) [inline src/wp-admin/includes/class-pclzip.php:4224]: Similar to my other comment, I'd like to suggest setting this to an empty string as well.

The below call to `gzinflate()` expects a string for the first parameter, so setting `$v_data` to `false` may cause other fatal errors in the future.

Ref: https://www.php.net/manual/en/function.gzinflate.php

File: src/wp-admin/includes/class-pclzip.php (hunk 1 above)
Comment (yakimun) [reply src/wp-admin/includes/class-pclzip.php:4215]: `fread` returns `false` on failure (https://www.php.net/manual/en/function.fread.php), so this code doesn't change anything except avoiding fatal error on PHP 8.

Comment (yakimun) [reply src/wp-admin/includes/class-pclzip.php:4215]: But I agree with you that an empty string looks better. I'll check that boolean values don't process especially and change `false` to `''`.

**Pull request closed without merging**